### PR TITLE
Feature/combine tiles

### PR DESCRIPTION
**Pull request by PhilippePoirier1998**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -22,21 +41,27 @@ fn main() {
...
-            Ok((nodes, edges)) => handle_output_format(fmt, nodes, edges, output),
-            Err(e) => println!("Error: {}", e),
-        }
+    // Convert the Vec<&str> into a HashSet<&str> for quicker searches
+    let ways_to_load_set: HashSet<&str> = ways_to_load_vec.into_iter().collect();
+    match loader::merge_csv_ways(way_files,&[output,"/way_properties.csv"].join(""), adjacentgeohashes.clone(), ways_to_load_set) {
+        Ok((f_records, nodes)) => {
+        merged_way_records=f_records;
+        nodes_to_load=nodes;},
+        Err(e) => println!("Error: {}", e),
```
Comment (ajasmin) [inline src/main.rs]: It may be hard to catch the error if we just print it...

File: src/osm4routing/loader.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        let file = File::open(filename)?;
+        let mut reader = csv::ReaderBuilder::new()
+        .has_headers(false) // Depending on your data, set this to true or false
+        .delimiter(b',')    // Set the custom delimiter as a byte
+        .from_reader(file);
+
+        for result in reader.records(){
+            let record = result?;
+            let id = record.get(0).unwrap_or("");
+            let node_list = record.get(1).unwrap_or("");
```
Comment (ajasmin) [inline src/osm4routing/loader.rs]: Does it make sense to use the empty string as a default, or should we panic instead?

**Review by ajasmin — APPROVED** [review]
Thanks!

**Comment by ajasmin** [comment]
You may want to try a pass of clippy to see if there are linter warnings

File: src/main.rs [hunk 3]
```diff
@@ -22,21 +44,41 @@ fn main() {
...
-        
-        match csv_read(args.get_str("<nodes.csv>"), args.get_str("<ways.csv>"))  {
-            Ok((nodes, edges)) => handle_output_format(fmt, nodes, edges, output),
-            Err(e) => println!("Error: {}", e),
-        }
+    let mut duration = start.elapsed();
+
+    println!("Parse input tile is: {:?}", duration);
+    start = Instant::now();
+    match loader::merge_csv_ways(way_files,&[output,"/way_properties.csv"].join(""), geohash_ptr.clone(), ways_to_load_set, adjacentgeohashes.clone()) {
```
Comment (PhilippePoirier1998) [inline src/main.rs]: change to merge msg_pack ways

**Pull request merged**